src/io/ipc/read/reader.rs

```diff
@@ -116,9 +116,9 @@ fn read_dictionary_block<R: Read + Seek>(
 pub fn read_file_dictionaries<R: Read + Seek>(
     reader: &mut R,
     metadata: &FileMetadata,
+    scratch: &mut ReadBuffer,
```

Comment:
This extra argument allows us to reuse the `ReadBuffer` that is owned by the `IPCFileReader`.